shap/explainers/_kernel.py

```diff
@@ -599,11 +598,12 @@ def solve(self, fraction_evaluated, dim):
         # do feature selection if we have not well enumerated the space
         nonzero_inds = np.arange(self.M)
         log.debug(f"{fraction_evaluated = }")
-        # if self.l1_reg == "auto":
-        #     warnings.warn(
-        #         "l1_reg=\"auto\" is deprecated and in the next version (v0.29) the behavior will change from a " \
-        #         "conditional use of AIC to simply \"num_features(10)\"!"
-        #     )
+        if self.l1_reg == "auto":
+            warnings.warn(
+                "l1_reg=\"auto\" is deprecated and in a future version the behavior will change from a " \
+                "conditional use of AIC to simply \"num_features(10)\"!",
+                DeprecationWarning
+            )
```

Comment:
If the explainer is called with the `__call__` interface, I don't think it's possible for users to control the value of `l1_reg`, so the warning will be impossible to fix. That's not necessarily an issue, but it may be a little annoying for users.

Would it be better to simply change the behaviour of "auto" to the desired new behaviour, as a breaking change? The fact that the regularisation behaviour will change in a future version has been declared in the docstring for several years now, so I think enough time has passed.

Reply:
I think in that case I would just set l1_reg directly to "num_features(10)" when going through the `__call__` interface. I would prefer this since it is more explicit.

Reply:
I think the `__call__` and `shap_values` interfaces should have the same default behaviour, so if we update one we might as well change the other. May I suggest changing the default behaviour as part of this PR?

Reply:
I would rather change the behaviour after some time so that users can adapt because changing defaults is a breaking change

Reply:
What is the alternative?
1. If the default of `__call__` is changed but, `shap_values` is not changed, they will have different behaviour which is probably rather unexpected and potentially confusing
2. If the defaults of both are unchanged, then users will get a DeprecationWarning with the "call" interface with no way to fix it

Reply:
I guess that is a good approach and raise a warning that says something like "The default value for `l1_reg=auto` is deprecated and will change in the future to `num_features(10)`. Please provide on of the following values ... to silence this warning."?

Reply:
Yes, let's do that.

Reply:
I'll merge this in as-is, and add the parameter in a follow-up PR...